coriolis/osmorphing/redhat.py

```diff
@@ -219,7 +219,9 @@ def _yum_clean_all(self):
     def _find_yum_repos(self, repos_to_enable=[]):
         found_repos = []
         for repo in repos_to_enable:
-            cmd = 'egrep "^\[.*%s.*\]$" -R /etc/yum.repos.d | cut -f2 -d:'
+            cmd = (
+                'egrep "^\[.*%s.*\]$" -R /etc/yum.repos.d | '
```

Comment:
This is a pretty underhanded way of getting the repos we want.

Presuming that a repo named `X` will always be in a file named `X.repo`, a more logical course of action imo would be to first search for `X.repo` in `/etc/yum.repos.d` (ideally through Python and not with `find` or `ls | grep`) and then process the file we found.

Presuming the repo can be in a file not named `X.repo`, it'd be better to first "parse" all the files in `/etc/yum.repos.d` and record all the repos and their files, then check for all the `repos_to_enable` you'll want in that.

Reply:
Your presumption is not necessarily correct. A `.repo` file can (and ususally does) store information for tens and hundreds of repos. Here we need to go through all the `.repo` files, fetch all declared repo names, and then check whether everything from `repos_to_enable` list is found.